### PR TITLE
Fix bug affecting algebra tactics

### LINKED ISSUE #436: `coq.ltac.call` fails with `Anomaly "Uncaught exception Not_found."`
Here is an [example](https://github.com/math-comp/algebra-tactics/pull/71#issuecomment-1321811531) by @proux01 (I didn't double-check):
```coq
From elpi Require Export elpi.
From mathcomp Require Import ssreflect ssrfun ssrbool eqtype.

Ltac ltac_foo := idtac.

Elpi Tactic fail_foo.
Elpi Accumulate lp:{{

pred solve i:goal, o:list sealed-goal.
solve (goal _ _ _ _ [_] as G) GS :-
  coq.ltac.call "ltac_foo" [] G GS.

}}.
Elpi Typecheck.

Goal True.
Proof.
elpi fail_foo ([the Equality.type of unit : Type]).
Qed.
```

This is a blocking issue in releasing Algebra Tactics compatible with Coq 8.17.

### PR DESCRIPTION
**Pull request by gares**
Fix #436

### CODE REVIEW

**Comment by pi8027** [comment]
@gares As a reminder, according to [what you said](https://github.com/math-comp/algebra-tactics/pull/71#issuecomment-1321828049), this PR may also fix this issue: https://coq.zulipchat.com/#narrow/stream/253928-Elpi-users-.26-devs/topic/Error.20.60Not.20a.20variable.20after.20goal.60

**Comment by gares** [comment]
I've reduced the bug so that is only affects tactics using `abstract`.

@pi8027 can you confirm this partial fix helps in your case?

I'll look also in the more general case of a tactic both calling abstract and leaving unresolved goals, but not today.

**Comment by pi8027** [comment]
Yes, let me check.

File: src/coq_elpi_HOAS.ml [hunk 1]
```diff
@@ -782,10 +780,18 @@ let show_coq_engine ?with_univs e = Format.asprintf "%a" (pp_coq_engine ?with_un
...
+   assert(env0 != env);
    let uctx = UState.update_sigma_univs (Evd.evar_universe_context sigma0) (Environ.universes env) in
-   let sigma = Evd.from_ctx (UState.demote_global_univs env uctx) in
-   from_env_sigma env sigma
+   let uctx = UState.demote_global_univs env uctx in
+   from_env_sigma env (Evd.set_universe_context sigma0 uctx)
+
+let from_env_keep_univ_of_sigma ~env0 ~env sigma0 =
+   assert(env0 != env);
+   let uctx = UState.update_sigma_univs (Evd.evar_universe_context sigma0) (Environ.universes env) in
```
Comment (gares) [inline src/coq_elpi_HOAS.ml:791]:
```suggestion
  assert(env0 != env);
  let uctx = UState.update_sigma_univs (Evd.evar_universe_context sigma0) (Environ.universes env) in
```

**Pull request merged**